Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Cancel.hs

```diff
@@ -88,10 +90,16 @@ cancel ::
   DM.Merchant ->
   SRB.Booking ->
   Maybe ST.SearchTry ->
-  Flow Bool
+  Flow (Bool, Maybe PriceAPIEntity)
 cancel req merchant booking mbActiveSearchTry = do
   CS.whenBookingCancellable booking.id $ do
     mbRide <- QRide.findActiveByRBId req.bookingId
+    cancellationCharge <- getCancellationCharges booking mbRide
+    whenJust cancellationCharge $ \fee -> do
+      riderId <- booking.riderId & fromMaybeM (BookingFieldNotPresent "rider_id")
```

Comment:
If riderId is not there, don't do anything instead of throw error.

Reply:
Done